logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -139,123 +140,270 @@ internal class CallDataSource(
+
+                        // Save into metadata
+                        updateCallMetadata(
+                            conversationId = conversationId,
+                            metadata = metadata
+                        )
+                    }
+                } else { // NOT IN CURRENT SESSION
+                    // Save into metadata
+                    updateCallMetadata(
+                        conversationId = conversationId,
+                        metadata = metadata
+                    )
+
+                    if (lastCallStatus in activeCallStatus) { // LAST CALL ACTIVE
+                        // Update database
+                        updateCallStatusById(
+                            conversationId = conversationId.toString(),
+                            status = CallStatus.CLOSED
+                        )
+                    }
+
+                    // Save into database
+                    wrapStorageRequest {
+                        callDAO.insertCall(call = callEntity)
+                    }
+                }
+            } else if (isGroupCall) {
+                if (isCallInCurrentSession) { // IN CURRENT SESSION
+                    if (lastCallStatus !in activeCallStatus) { // LAST CALL IS NOT ACTIVE
+                        // Save into metadata
+                        updateCallMetadata(
+                            conversationId = conversationId,
+                            metadata = metadata
+                        )
+
+                        // Save into database
+                        wrapStorageRequest {
+                            callDAO.insertCall(call = callEntity)
+                        }
+                    }
+                } else { // NOT IN CURRENT SESSION
+                    // Save into metadata
+                    updateCallMetadata(
+                        conversationId = conversationId,
+                        metadata = metadata
+                    )
+
+                    if (lastCallStatus in activeCallStatus) { // LAST CALL IS ACTIVE
+                        // Update database
+                        updateCallStatusById(
+                            conversationId = conversationId.toString(),
+                            status = CallStatus.STILL_ONGOING
+                        )
+                    } else { // LAST CALL NOT ACTIVE
+                        // Save into database
+                        wrapStorageRequest {
+                            callDAO.insertCall(call = callEntity)
+                        }
+                    }
+                }
+            }
+        } else if (status == CallStatus.STARTED) {
+            // Save into database
+            wrapStorageRequest {
+                callDAO.insertCall(call = callEntity)
+            }
+
+            // Save into metadata
+            updateCallMetadata(
+                conversationId = conversationId,
+                metadata = metadata
+            )
+        }
```

Comment:
To be honest all of the code inside `createCall` inside `CallRepository` must be reworked inside createCall and IncomingCall usecases as it will hold a better power of the logic there.

Although this task has been ongoing for the past 3 weeks, and to not hold it anymore we can leave it as it is and work on the changes on it in a later task.

Reply:
yes makes sense..but it would be better if we merge non duplicated code to develop, and later we can improve it more